LethalAPI.Core/Plugin.cs

```diff
@@ -10,32 +10,58 @@ namespace LethalAPI.Core;
 // ReSharper disable InconsistentNaming
 // ReSharper disable MemberCanBePrivate.Global
 #pragma warning disable SA1401 // field should be made private
+using System;
 
 using BepInEx;
 using BepInEx.Logging;
 using HarmonyLib;
+using MEC;
 
 /// <inheritdoc />
 [BepInPlugin(PluginInfo.PLUGIN_GUID, PluginInfo.PLUGIN_NAME, PluginInfo.PLUGIN_VERSION)]
 public class Plugin : BaseUnityPlugin
 {
+    /// <summary>
+    /// Gets the singleton for a plugin.
+    /// </summary>
+    public static Plugin Singleton = null!;
+
+    /// <summary>
+    /// Gets the <see cref="BepInEx.Logging.Logger"/>.
+    /// </summary>
     /// <summary>
     /// The base logger.
     /// </summary>
-    internal static ManualLogSource Log;
+#pragma warning disable SA1309
```

Comment:
What are these pragmas for exactly?

Reply:
A field name in C# begins with an underscore.
This needs to be moved to the top of the class, as it's cleaner. I will document it as well

Reply:
Also the field has to be _Logger because a) Logger is already in the base class, therefore cannot be safely overriden, and B) _Logger is the actual static instance. (It is only used by the Log class.)